### DISCUSSION TITLE
Apple signin suddenly broke for me

### DISCUSSION THREAD

**Discussion by 98mux** [pocketbase/pocketbase#5888] [Q&A] [answered]
Hi there

I'm using authWithOAuth2 from the web package.

Suddenly apple signin broke for me without changing any code.
My web pocketbase version number is 0.21.5 and my pocketbase instance version is 0.22.19.

It broke 3 days ago, this date here was the last time it successfully worked "2024-11-14 00:36:00"

So I know this happened without a code change because my app can't instantly update it has to go through the app store review. Also it broke suddenly on the same app version. I also haven't touched the pocketbase instance oauth settings.

The network call the authWithOAuth2 function uses is broken for me:

https://instance-pocketbase.up.railway.app/api/collections/users/auth-with-oauth2
POST
400 Bad Request

Payload:
{code: "<removed>"
codeVerifier: "<removed>"
provider: "apple"
redirectUrl: "https://instance-pocketbase.up.railway.app/api/oauth2-redirect"}

There might be some credential issue or something, or maybe something is expired. I've looked at my keys and I haven't seen anything obvious. I'll go through them again and double check. Maybe I should redo the apple client secret, might be worth a shot.

Note: Google signin is still working

Has anyone else experienced this?

**Comment by 98mux** [comment, marked as answer]
The client secret did indeed expire. I saw that the default expiration date was 6 months within pocketbase - which is a time length that makes sense for it to have expired.

Maybe a better error message for expired tokens? Or some error messages or warnings showing up within the pocketbase interface would be cool too (took me 3 days to notice it)

**Reply by ganigeorgiev** [reply]
The 6 months expiration is an Apple-only thing.

If you think that the flow could be improved feel free to create a PR but the Apple integration already differ so much from the others and I really want to avoid having another special behavior just because it is Apple.
